resources/migrations/000_migrations.yaml

```diff
@@ -14812,6 +14812,47 @@ databaseChangeLog:
                   defaultValue: null
                   remarks: 'This is used to differentiate instance-analytics collections from all other collections.'
 
+  - changeSet:
+      id: v47.00-035
+      author: calherries
+      comment: Added 0.47.0 - Drop foreign key constraint on implicit_action.action_id
+      changes:
+        - dropForeignKeyConstraint:
+            baseTableName: implicit_action
+            constraintName: fk_implicit_action_action_id
+      rollback:
+        - addForeignKeyConstraint:
+            baseTableName: implicit_action
+            baseColumnNames: action_id
+            referencedTableName: action
+            referencedColumnNames: id
+            constraintName: fk_implicit_action_action_id
+            onDelete: CASCADE
+
+  - changeSet:
+      id: v47.00-036
+      author: calherries
+      comment: Added 0.47.0 - Set primary key to action_id for implicit_action table
+      changes:
+        - addPrimaryKey:
+            tableName: implicit_action
+            columnNames: action_id
+            constraintName: pk_implicit_action
+
+  - changeSet:
+      id: v47.00-037
+      author: calherries
+      comment: Added 0.47.0 - Add foreign key constraint on implicit_action.action_id
```

Comment:
do you think we need an index?

Reply:
Indices are automatically created for primary keys in postgres, mysql, and h2.
Postgres: https://www.postgresql.org/docs/current/indexes-unique.html#:~:text=PostgreSQL%20automatically%20creates%20a%20unique,mechanism%20that%20enforces%20the%20constraint.
MySQL: https://www.tutorialspoint.com/is-the-primary-key-automatically-indexed-in-mysql#:~:text=Yes%2C%20primary%20key%20is%20automatically,primary%20key%20to%20be%20indexed.
H2: https://groups.google.com/g/h2-database/c/Hi_JnOPM9vI?pli=1

Reply:
this is a FK right? FKs are not automatically indexed https://github.com/metabase/metabase/issues/30924

Reply:
It's a PK. See the migration above https://github.com/metabase/metabase/pull/31292/files#diff-241462177925c37373615bb7d440f151756dc4e629f4fbd5a6ac26bf6a688d4fR14835